services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowCrudSpec.groovy

```diff
@@ -370,6 +375,42 @@ class FlowCrudSpec extends BaseSpecification {
         ]
     }
 
+    def "Unable to create a flow with asymmetric forward and reverse paths"() {
+        given: "Two active switches and two possible flow paths at least"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = db.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            possibleFlowPaths.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Make the shorter forward path not preferable"
+        pathHelper.getInvolvedIsls(possibleFlowPaths[0]).each { db.updateLinkCost(it, Integer.MAX_VALUE) }
+
+        and: "Make the longer reverse path not preferable"
```

Comment:
So we have cheap longer forward path and cheap short reverse path, correct?

Reply:
Cleanup is done at L411.